### PR TITLE
External References fields

### PR DESCRIPTION
**Pull request by baytelman**
Example: 
User > SMS
Stream > Message

### CODE REVIEW

File: src/AutoAdmin.tsx [hunk 1]
```diff
@@ -49,6 +50,13 @@ interface AutoAdminAttribute {
   fieldOptions?: any;
 }
 
+interface AutoAdminReference {
```
Comment (baytelman) [inline src/AutoAdmin.tsx]: Reference another table

File: src/AutoAdmin.tsx [hunk 2]
```diff
@@ -78,6 +86,11 @@ const enumToChoices = (e: any) => Object.keys(e).map((key: string) => ({ id: e[k
 const attributeToField = (input: AutoAdminAttribute) => {
   if (Array.isArray(input.type) && input.type.length > 0) {
     const inputType: string | AutoAdminAttribute = input.type[0];
+
+    /* Force the label to be the attribute name */
+    if (!input.label) {
+      input.label = input.attribute;
```
Comment (baytelman) [inline src/AutoAdmin.tsx]: If no label, use the attribute name

File: src/AutoAdmin.tsx [hunk 3]
```diff
@@ -220,12 +244,23 @@ const tabbedForm = (schema: AutoAdminAttribute[]) => {
   );
 };
 
-const tabbedLayout = (schema: AutoAdminAttribute[]) => {
+const referenceTab = (reference: AutoAdminReference) => {
```
Comment (baytelman) [inline src/AutoAdmin.tsx]: Place references (to external data) in another tab

File: src/AutoAdmin.tsx [hunk 4]
```diff
@@ -263,24 +301,43 @@ export const AutoEdit = (props: any, { schema }: { schema: AutoAdminAttribute[]
   );
 };
 
-export const AutoList = (props: any, { schema }: { schema: AutoAdminAttribute[] }) => {
+export const AutoDataGrid = (props: any, { schema }: { schema: AutoAdminAttribute[] }) => {
```
Comment (baytelman) [inline src/AutoAdmin.tsx]: This is how we show external references.

File: lib/AutoAdmin.d.ts [hunk 5]
```diff
@@ -9,20 +9,31 @@ interface AutoAdminAttribute {
     readOnly?: boolean;
     fieldOptions?: any;
 }
+interface AutoAdminReference {
+    reference: string;
```
Comment (baytelman) [inline lib/AutoAdmin.d.ts]: external reference (i.e. "message")

File: lib/AutoAdmin.d.ts [hunk 6]
```diff
@@ -9,20 +9,31 @@ interface AutoAdminAttribute {
...
+    target: string;
```
Comment (baytelman) [inline lib/AutoAdmin.d.ts]: how to get to that field (i.e. "user" as in "message.user")

File: lib/AutoAdmin.d.ts [hunk 7]
```diff
@@ -9,20 +9,31 @@ interface AutoAdminAttribute {
...
+    tab?: string;
+    schema: AutoAdminAttribute[];
```
Comment (baytelman) [inline lib/AutoAdmin.d.ts]: List of fields to be displayed in the reference a table.

File: demo/src/index.tsx [hunk 8]
```diff
@@ -209,7 +209,10 @@ const USER_SCHEMA = [
 const TagAdmin = AutoResource('tags', { schema: TAG_SCHEMA });
 const PostAdmin = AutoResource('posts', { schema: POST_SCHEMA });
 const UserAdmin = AutoResource('users', {
-  schema: USER_SCHEMA
+  schema: USER_SCHEMA,
+  references: [
+    {reference: 'posts', target: 'author', schema: POST_SCHEMA}
```
Comment (baytelman) [inline demo/src/index.tsx]: In the demo, the USER displays all her/his posts.

File: src/AutoAdmin.tsx (hunk 2 above)
Comment (golastra) [reply src/AutoAdmin.tsx]: If no label leave it empty?

File: lib/AutoAdmin.d.ts (hunk 5 above)
Comment (golastra) [reply lib/AutoAdmin.d.ts]: `react-admin` already supports external references thru endpoints. Does this replace that feature, so users don't have to implement new ones and directly access what they need?

File: src/AutoAdmin.tsx (hunk 2 above)
Comment (baytelman) [reply src/AutoAdmin.tsx]: If no label, we use the attribute name by default.

File: lib/AutoAdmin.d.ts (hunk 5 above)
Comment (baytelman) [reply lib/AutoAdmin.d.ts]: It uses the internal implementation. This is a quick way to say "what schema (what columns) you want to show")

**Pull request merged**